codegen/java-gen/src/main/scala/kalix/codegen/java/WorkflowSourceGenerator.scala

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright 2024 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.codegen.java
+
+import kalix.codegen.Format
+import kalix.codegen.ModelBuilder
+import kalix.codegen.ProtoMessageType
+import kalix.codegen.SourceGeneratorUtils.allRelevantMessageTypes
+import kalix.codegen.SourceGeneratorUtils.collectRelevantTypes
+import kalix.codegen.SourceGeneratorUtils.generateImports
+import kalix.codegen.SourceGeneratorUtils.lowerFirst
+import kalix.codegen.SourceGeneratorUtils.managedComment
+import kalix.codegen.SourceGeneratorUtils.qualifiedType
+import kalix.codegen.SourceGeneratorUtils.unmanagedComment
+import kalix.codegen.java.JavaGeneratorUtils.typeName
+import kalix.codegen.java.JavaGeneratorUtils.writeImports
+
+object WorkflowSourceGenerator {
+
+  private[codegen] def workflowSource(
+      service: ModelBuilder.EntityService,
+      workflowComponent: ModelBuilder.WorkflowComponent,
+      packageName: String,
+      className: String): String = {
+
+    val imports = generateImports(
+      allRelevantMessageTypes(service, workflowComponent),
+      packageName,
+      otherImports = Seq("kalix.javasdk.workflow.WorkflowContext"))
+
+    val stateType = workflowComponent.state.messageType.fullName
+
+    val methods = service.commands.map { cmd =>
+      val methodName = cmd.name
+      val inputType = cmd.inputType.fullName
+      val outputType = qualifiedType(cmd.outputType)
+
+      s"""|@Override
+          |public Effect<$outputType> ${lowerFirst(methodName)}($stateType currentState, $inputType ${lowerFirst(
+        cmd.inputType.name)}) {
+          |  return effects().error("The command handler for `$methodName` is not implemented, yet");
+          |}
+          |""".stripMargin
+    }
+
+    s"""package $packageName;
+       |
+       |${writeImports(imports)}
```

Comment:
It's pity that we didn't refactored the java codegen to use our `c` interpolator. Check out the scala codegen for more details.

It's much easier to use the `c` interpolator and then we get the imports right out of the box.

We should do some gardening in the SDK to solve this kind of issues.

Reply:
https://github.com/lightbend/kalix-jvm-sdk/issues/1991